core/src/main/scala/quasar/datasource/kafka/Config.scala

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2020 Precog Data
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package quasar.datasource.kafka
+
+import slamdata.Predef._
+
+import argonaut.Argonaut._
+import argonaut._
+import cats.data.NonEmptyList
+import cats.syntax.either._
+import quasar.connector.DataFormat
+
+case class Config(
+  bootstrapServers: NonEmptyList[String],
+  groupId: String,
+  topics: NonEmptyList[String],
+  decoder: Decoder,
+  format: DataFormat
+) extends Product with Serializable {
+  def isTopic(topic: String): Boolean = topics.exists(_ == topic)
+
+  // No sensitive information at this time
+  def sanitize: Config = this
+  def reconfigure(patch: Config): Either[Config, Config] = patch.asRight
+}
+
+object Config {
+
+  implicit def nelEJ[A: EncodeJson]: EncodeJson[NonEmptyList[A]] = _.toList.asJson
+
+  implicit def nelDJ[A: DecodeJson]: DecodeJson[NonEmptyList[A]] = c => c.as[List[A]].flatMap {
```

Comment:
Can we use the `DecodeJson` that [argonaut provides](https://github.com/argonaut-io/argonaut/blob/f84fbfafcab0126a2cca0e2bb276a63fd7104d5a/argonaut-cats/shared/src/main/scala/argonaut/DecodeJsonCats.scala#L9)?

Reply:
The error message produced by Argonaut's `DecodeJson[NonEmptyList[A]]` is very unhelpful. I found it after I had created the above, and promptly switched to it only to discover there was no mention of the field in the message (only on history), nor that the problem was the json array was empty.

Reply:
👍 can you add a comment to that effect here? That way we'll remember in the future why we chose to use our own codec.